org/mozilla/jss/pkcs11/PK11Signature.c

```diff
@@ -31,12 +32,18 @@ getSomeKey(JNIEnv *env, jobject sig, void **key, short type);
 static SECOidTag
 getAlgorithm(JNIEnv *env, jobject sig);
 
+static SECOidTag
+getDigestAlgorithm(JNIEnv *env, jobject sig);
+
+
 static void
 setSigContext(JNIEnv *env, jobject sig, jobject context);
 
 static PRStatus
 getSigContext(JNIEnv *env, jobject sig, void**pContext, SigContextType* pType);
 
+static SECAlgorithmID *getRSAPSSParamsAndSigningAlg(PRArenaPool *arena,JNIEnv *env, jobject this,SECKEYPrivateKey *privk); 
```

Comment:
nit: I'd prefer `env, this, arena, privk` rather than `arena, env, this, privk` -- `env, this` pattern is typically used from the `native` stubs JNI produces and is always first. I think grouping Java and then NSS arguments makes the most sense to me. 

And one other thing to consider: what do the failure cases in `getRSAPSSParamsAndSigningAlg` look like? Would

```c
static SECStatus getRSAPSSParamsAndSigningAlg(JNIEnv *env, jobject this, PRArenaPool *arena, SECAlgorithmId *alg, SECKEYPrivateKey privk);
```

be a better signature? Then you can do (below):

```c
SECAlgorithmId signAlg;
if (getRSAPSSParamsAndSigningAlg(env, this, arena, &signAlg, privk) != PR_SUCCESS) {
    goto finish;
}

ctx = SGN_NewContextWithAlgorithmID(&signAlg, privk);
```

and the intent is more clear IMO.

Reply:
I took your advice mostly.. Done. Thx.